consensus/src/round_manager.rs

```diff
@@ -1222,6 +1224,47 @@ impl RoundManager {
         result
     }
 
+    async fn new_qc_from_order_vote_msg(
+        &mut self,
+        order_vote_msg: &OrderVoteMsg,
+    ) -> anyhow::Result<()> {
+        ensure!(
+            order_vote_msg
+                .order_vote()
+                .ledger_info()
+                .consensus_block_id()
+                == order_vote_msg.quorum_cert().certified_block().id(),
+            "QuorumCert attached to order vote doesn't match"
+        );
+
+        if let NeedFetchResult::QCAlreadyExist = self
+            .block_store
+            .need_fetch_for_quorum_cert(order_vote_msg.quorum_cert())
+        {
+            return Ok(());
```

Comment:
we use the qc from the order vote message in `process_order_vote_reception_result` but it's unverified. (it's unlikely a problem since it'll hit QCAlreadyExist too but feels fragile)

Reply:
I think we can change the later process (`new_ordered_cert`) to not include the QC at all, since the QC is already being added from this function.

Reply:
Before calling `process_order_vote_reception_result`, we are calling `new_qc_from_order_vote_msg` which would store QC in the block store if not already present. So, we wouldn't need to store QC again in `process_order_vote_reception_result`. So, I removed storing QC again in `process_order_vote_reception_result`.